src/components/List/List.jsx

```diff
@@ -1,35 +1,74 @@
-import React, { useRef, forwardRef, useMemo } from "react";
+import React, { useRef, forwardRef, useMemo, useState, useCallback } from "react";
 import PropTypes from "prop-types";
 import cx from "classnames";
 import useMergeRefs from "../../hooks/useMergeRefs";
 import { VirtualizedListItems } from "components/List/VirtualizedListItems/VirtualizedListItems";
 import "./List.scss";
+import { keyCodes } from "../../constants/KeyCodes";
 
 const List = forwardRef(
   ({ className, id, component, children, dense, ariaLabel, ariaDescribedBy, renderOnlyVisibleItems, style }, ref) => {
     const componentRef = useRef(null);
+    const [focusIndex, setFocusIndex] = useState(0);
     const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
     const Component = component;
+    const childrenRefs = useRef([]);
+    const changeFocus = focusIndex => {
+      setFocusIndex(focusIndex);
+      childrenRefs.current[focusIndex].focus();
+    };
+    const onKeyDown = useCallback(
+      event => {
+        const isUpKey = event.keyCode === keyCodes.UP_ARROW;
+        const isDownKey = event.keyCode === keyCodes.DOWN_ARROW;
+        if (isUpKey || isDownKey) {
+          if (isDownKey && focusIndex + 1 < childrenRefs.current.length) {
+            changeFocus(focusIndex + 1);
+          } else if (isUpKey && focusIndex > 0) {
+            changeFocus(focusIndex - 1);
+          }
+          event.preventDefault();
+        }
+      },
+      [focusIndex]
+    );
+    const childMapping = useCallback(
+      (child, index) => {
+        return {
+          ...child,
+          ref: ref => (childrenRefs.current[index] = ref),
+          props: {
+            ...child.props,
+            tabIndex: focusIndex === index ? "0" : "-1"
+          }
+        };
+      },
+      [focusIndex]
+    );
     const overrideChildren = useMemo(() => {
       let override = children;
       if (renderOnlyVisibleItems) {
         override = <VirtualizedListItems>{override}</VirtualizedListItems>;
+      } else {
+        override = override.map(childMapping);
```

Comment:
This is nice :) But not very standard. Also it will not work in case you have single element/null/undef as child (never should be the case in a list but still we do not want to throw an error.)
Please use React.Children.map for iterate the children and React.cloneElement for cloning the element (this way you also will not need to use useCallback for child mapping)